### DISCUSSION TITLE
tp1-ejercicio11

### DISCUSSION THREAD

**Discussion by Clirussa** [INGCOM-UNRN-P1/cursada-2023#112] [Preguntas y respuestas] [answered]
Hola, estamos (al fin) terminando el TP 1 con @agosgit  . Tenemos un problema y es que cuando imprime, en lugar de 0 en el límite, sale un 2 que no entendemos de donde sale o como se puede modificar. 

Esperamos ayuda
Muchas gracias!

![ejercicio11-](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/102642611/ea6a8fcf-5d55-4101-acaf-f372b70b23cc)
![ejercicio 11-2](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/102642611/7ed043aa-0ee8-42cb-b6b4-ae0b416b2f78)

**Comment by martinvilu** [comment]
Las funciones, a no ser de que proposito explicito sea el mostrar u obtener información del usuario; no van con `printf` o `scanf`, respectivamente.

¿Qué devuelve `printf`?

**Reply by Clirussa** [reply]
Devuelve los valores que va tomando i. Si no nos aparece todo 0.

**Reply by martinvilu** [reply, marked as answer]
Tené en cuenta que nunca usamos `printf` como ‘r-value’; el retorno de la función no es lo que esperas.

Esta es otra razón para no usar la función dentro de las funciones que resuelven los ejercicios

**Reply by Clirussa** [reply]
Si, cuando me.lo marcaste me acordé. ok. ahora corregimos. Gracias.

**Comment by martinvilu** [comment]
El contador circular, debe retornar `i` incrementado en uno. Cuando sea igual a `limite` la función retornará 0.
